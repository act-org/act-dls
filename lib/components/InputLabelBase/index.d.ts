/**
 * @prettier
 */
import * as React from 'react';
import { InputLabelProps } from '@material-ui/core';
export declare type Props = InputLabelProps;
declare const InputLabelBase: React.FC<Props>;
export default InputLabelBase;
//# sourceMappingURL=index.d.ts.map