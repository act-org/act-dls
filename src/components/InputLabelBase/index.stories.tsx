/**
 * @prettier
 */

import * as React from 'react';
import { boolean, select, text } from '@storybook/addon-knobs';

import InputLabelBase from '.';

export const Base = (): React.ReactElement<any> => (
  <InputLabelBase
    color={select(
      'Color',
      {
        Primary: 'primary',
        Secondary: 'secondary',
      },
      'primary',
    )}
    disabled={boolean('Disabled?', false)}
    error={boolean('Error?', false)}
    focused={boolean('Focused?', false)}
    required={boolean('Required?', false)}
    variant={select(
      'Variant',
      {
        Filled: 'filled',
        Outlined: 'outlined',
        Standard: 'standard',
      },
      'standard',
    )}
  >
    {text('Label', 'Input Label')}
  </InputLabelBase>
);

export default {
  component: InputLabelBase,
  parameters: {
    info: {
      text: 'This is the base variant of the InputLabel component.',
    },
  },
  title: 'Atoms/InputLabel',
};
