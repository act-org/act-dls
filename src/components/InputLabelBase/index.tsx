/**
 * @prettier
 */

import * as React from 'react';
import { InputLabel, InputLabelProps } from '@material-ui/core';

export type Props = InputLabelProps;

const InputLabelBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <InputLabel {...props} />;

export default InputLabelBase;
