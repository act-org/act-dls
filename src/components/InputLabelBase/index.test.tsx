/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import InputLabelBase from '.';

describe('InputLabelBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <InputLabelBase color="primary" required>
        InputLabel
      </InputLabelBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
