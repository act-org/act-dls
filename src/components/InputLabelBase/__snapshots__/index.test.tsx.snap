// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabelBase matches the snapshot 1`] = `
<div>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated Mui-required Mui-required"
  >
    InputLabel
    <span
      aria-hidden="true"
      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
    >
      â€‰
      *
    </span>
  </label>
</div>
`;
